src/main/java/com/gtnewhorizons/angelica/transform/HUDCachingTransformer.java

```diff
@@ -0,0 +1,75 @@
+package com.gtnewhorizons.angelica.transform;
+
+import com.gtnewhorizons.angelica.loading.AngelicaTweaker;
+import net.minecraft.launchwrapper.IClassTransformer;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AnnotationNode;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.InsnList;
+import org.objectweb.asm.tree.InsnNode;
+import org.objectweb.asm.tree.JumpInsnNode;
+import org.objectweb.asm.tree.LabelNode;
+import org.objectweb.asm.tree.LdcInsnNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+
+
+public class HUDCachingTransformer implements IClassTransformer {
+    private static final String HUDCaching = "com/gtnewhorizons/angelica/hudcaching/HUDCaching$Hooks";
+
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] basicClass) {
+
```

Comment:
you need a if(basiclass == null) return null;

sometimes the classloading can fail and a null class bytes will arrive here